Scripts/Source/minai_Arousal.psc

```diff
@@ -415,3 +515,24 @@ string Function GetFactionsForActor(actor akTarget)
   return ret
 EndFunction
 
+
+bool Function IsTNGExposed(Actor akActor)
+    int itemIndex = akActor.GetNumItems()
```

Comment:
This is going to be really nonperformant for people with lots of items in their inventory - This will put a lot of load on the scripting engine. Let's find a better way to do this.

Reply:
It looks like all you're doing is seeing if they have a cuirass equipped, and then seeing if the armor has those keywords on it. Try something like this instead:
```
bool Function IsTNGExposed(actor akActor)
  Armor armorItem = akActor.GetWornForm(cuirassSlot)
  if armorItem
    ; The rest of your logic
  EndIf
EndFunction
```